icu4j/main/classes/core/src/com/ibm/icu/text/PluralRules.java

```diff
@@ -1276,14 +1229,21 @@ private static void checkDecimal(SampleType sampleType2, FixedDecimal sample) {
          * @deprecated This API is ICU internal only.
          */
         @Deprecated
-        public Set<Double> addSamples(Set<Double> result) {
-            for (FixedDecimalRange item : samples) {
-                // we have to convert to longs so we don't get strange double issues
-                long startDouble = item.start.getShiftedValue();
-                long endDouble = item.end.getShiftedValue();
+        public Set<DecimalQuantity> addSamples(Set<DecimalQuantity> result) {
```

Comment:
Suggestion: *Unless* you ever need to round-trip the return values, you should return `Set<BigDecimal>`, and bubble that change through to other APIs that depend on this one. This would clean up the code in this function a lot, and it is more consistent with what we had here before.

If you need to round-trip the return value, please point me to where you need that behavior and let's figure something out.